features/environment.py

```diff
@@ -412,6 +415,58 @@ def before_scenario(context: Context, scenario: Scenario):
             )
 
 
+def after_scenario(context, scenario):
+    """Collect the coverage files after the scenario is run."""
+    if context.pro_config.collect_coverage:
+        cov_dir = os.path.join(context.pro_config.artifact_dir, "coverage")
+        if not os.path.exists(cov_dir):
+            os.makedirs(cov_dir)
+
+        inner_dir = os.path.basename(scenario.filename.replace(".", "_"))
+        new_artifacts_dir = os.path.join(
+            cov_dir,
+            inner_dir,
+        )
+        if not os.path.exists(new_artifacts_dir):
+            os.makedirs(new_artifacts_dir)
+        if hasattr(context, "machines") and SUT in context.machines:
+            try:
+                scenario_name = (
+                    os.path.basename(scenario.filename.replace(".", "_"))
+                    + "_"
+                    + str(scenario.line)
+                )
+                cov_filename = ".coverage.{}".format(scenario_name)
```

Comment:
Any reason why the coverage file needs to start with a dot ?

Reply:
I think it's naming convention followed universally for coverage files to keep them hidden when generated so it does not clutter the user's file system given it is only used by developers for internal usage. Also, the python coverage library by default looks for .coverage files